SYCL/Basic/regression/build_log.cpp

```diff
@@ -0,0 +1,156 @@
+// REQUIRES: gpu
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.exe
+// RUN: %GPU_RUN_PLACEHOLDER %t.exe 2> %t.out || true
+
+// RUN: FileCheck %s --check-prefix=CHECK-EXPECTED-ERROR --input-file %t.out
+// CHECK-EXPECTED-ERROR: Total size of kernel arguments exceeds limit! Total arguments size: 1056, limit: 1024
```

Comment:
1. Do not see any prints in the test itself, which component does print this message( and where )?
2. I believe the content of the message is produced by IGC, so it worries me that we test message that we do not control, I would even say that the behavior which is tested here has nothing to do with SYCL.

Reply:
Yes. The message is coming from IGC, and the intent of the test is that we are able to print an error to DPC++ users. if the IGC message changes then we will need to fix the test.

Reply:
Can't we check more "standard string"(like "build failed) in the log message, I think it would make test more robust.

Reply:
We could, of course, but it would not test the ability to transport native build-log messages to DPC++ users.

Reply:
No, I suggest checking "native" build log message. Usually this message start with something like: "compilation failed". Let's discuss offline.

Reply:
Ah, ok. The full message from IGC seems to be this:
~~~
error: Total size of kernel arguments exceeds limit! Total arguments size: 1056, limit: 1024
error: backend compiler failed build.
~~~
I will change to test the "error: backend compiler failed build" substring.